src/preamble.js

```diff
@@ -139,11 +139,178 @@ function updateMemoryViews() {
 #if SUPPORT_BIG_ENDIAN
   Module['HEAP_DATA_VIEW'] = HEAP_DATA_VIEW = new DataView(b);
 #endif
+#if MEMORY64 && MAXIMUM_MEMORY > FOUR_GB
+  // Chrome does not allow TypedArrays with more than 4294967296 elements
+  // We'll create proxy objects for HEAP(U)8 when memory is > 4gb and for HEAP(U)16 when > 8gb
+  // https://bugs.chromium.org/p/v8/issues/detail?id=4153
+  var maxArraySize = Math.min(b.byteLength, 4 * 1024 * 1024 * 1024 - 8);
+  /**
+   * @param {string} type - Heap type
+   * @param {number} [offset] - Heap offset
+   * @param {number} [length] - typed array length
+   */
+  const getHeapBlock = (type, offset, length) => {
```

Comment:
How about using `function getHeapBlock(type, offset, length) {` here for `proxyHandler` .. to match c`reateMemoryProxy`?